src/interfaces/chain.h

```diff
@@ -133,12 +133,23 @@ class Chain
         int64_t* max_time = nullptr,
         int64_t* mtp_time = nullptr) = 0;
 
+    //! Find ancestor of block at specified height and return its hash.
+    virtual uint256 findAncestorByHeight(const uint256& block_hash, int ancestor_height) = 0;
+
     //! Return whether block descends from a specified ancestor, and
     //! optionally return height of the ancestor.
     virtual bool findAncestorByHash(const uint256& block_hash,
         const uint256& ancestor_hash,
         int* height = 0) = 0;
 
+    //! Find most recent common ancestor between two blocks and optionally
+    //! return its hash and/or height. Also return height of first block. Return
```

Comment:
I think it's confusing to return something unrelated to the common ancestor here...

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17954#discussion_r381473300

> I think it's confusing to return something unrelated to the common ancestor here...

Agreed, will backport 25c1ae48204215622bc9fd3a8bc9677f15c32674 ([branch](https://github.com/ryanofsky/bitcoin/commits/pr/getblock)) as soon as I get a chance